onedrive-authorize

```diff
@@ -9,7 +9,7 @@ echo "https://login.live.com/oauth20_authorize.srf?client_id=${api_client_id}&sc
 echo ""
 echo "When ready, please enter the value of the code parameter (from the URL of the blank page) and press return"
 read code
-api_json_result=$(curl -s -d "client_id=${api_client_id}&client_secret=${api_client_secret}&code=${code}&grant_type=authorization_code&redirect_uri=https://login.live.com/oauth20_desktop.srf" -X POST "${api_auth_url}")
```

Comment:
According to the official documentation, the parameter `client_secret` is required: https://dev.onedrive.com/auth/msa_oauth.htm#step-2-redeem-the-code-for-access-tokens

Even though it works without providing the secret, I would leave it there until Microsoft changes the documentation.